test/controllers/users_controller_test.rb

```diff
@@ -1,41 +1,28 @@
+# frozen_string_literal: true
+
 require "test_helper"
 
+# Test user controller
+# extracted from master ObserverControllerTest
 class UsersControllerTest < FunctionalTestCase
 
   # ------------------------------------------------------------
-  #  User
-  #  User_controller
+  # indexes / searches
   # ------------------------------------------------------------
 
-  def test_show_user_no_id
-    get(:show)
-    assert_redirected_to(action: :index_user)
-  end
-
-  def test_some_admin_pages
-    [
-      [:users_by_name,  "list_users", {}],
-      [:email_features, "email_features", {}]
-    ].each do |page, response, params|
-      logout
-      get(page, params: params)
-      assert_redirected_to(controller: :account, action: :login)
+  # Prove that user_index is restricted to admins
+  def test_index_user
+    login("rolf")
+    get(:index_user)
 
-      login("rolf")
-      get(page, params: params)
-      assert_redirected_to(controller: :rss_logs, action: :index)
-      assert_flash_text(/denied|only.*admin/i)
+    # assert_redirected_to(:root)
+    assert_redirected_to(:rss_logs)
```

Comment:
This fixes a bug caused by changing `:root` between `master` and `bootstrap-4`; it was `:rss_logs`, now is `observations#index`.
I'm concerned about that change. `master` gives users the option of seeing updates for all objects or only some, e.g., just Observations (though it's a little hard to notice that option). `bootstrap-4` seems to remove those options. (I, and some other users, want to see them all.)

Reply:
Hi Joe - I anticipated that and discussed a bit with Jason a few months ago.

Note that in routes.rb, :root actually has two behaviors: it's observations#index for non-logged-in users, and rss_logs#index for logged-in users. 

To me this makes more sense - a lurker likely wants the observation log only, and either will be confused by or have no interest in the rss_log. In any case they cannot do much with anything in rss_log, unless logged in. Because the site remembers logins, regular users should notice no difference - they will be directed to rss_logs#index as the :root

Of course I want to run that by everyone, but I think people will agree that as a workflow it feels "friendlier" to approach newbies with a page that is intelligible - rather than our full "kitchen sink".

Reply:
Therefore, to the question of your bugfix: Logged in users should probably be redirected to :root after all, for this action. 
Let's fix routes with a small PR now?

Reply:
@nimmolo: Thanks. I hadn't noticed the different behavior for non-logged in vs logged-in. (I was lazy, and looked at the text generated by `rails routes`. rather than the routes themselves.)
It makes good sense.
Take your time in uncommenting that line. It might be counter-productive to uncomment it until you've finished templating the changes for rss_logs#index.